app/scripts/services/secrets.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("SecretsService", function($q, $filter, DataService){
+
+    var newestSecret;
+
+    var setNewestSecretName = function(secretName) {
+      newestSecret = secretName;
+    };
+
+    var getNewestSecretName = function() {
+       return newestSecret;
+    };
```

Comment:
Just a note here that I think we can remove "newest secret." See my comments above.